proxy/inetaf_tcpproxy.go

```diff
@@ -0,0 +1,104 @@
+package proxy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+
+	"github.com/inetaf/tcpproxy"
+)
+
+type childProxy struct {
+	l net.Listener
+	s Server
+}
+
+type InetAfTCPProxyServer struct {
+	Proxy    *tcpproxy.Proxy
+	children []*childProxy
+}
+
+// Close - implements Server - is this even called?
+func (tps *InetAfTCPProxyServer) Close() error {
+	_ = tps.Proxy.Close()
+	firstErr := tps.Proxy.Wait()
+	errChan := make(chan error)
+	for _, sl := range tps.children {
+		go func(sl *childProxy) {
+			errChan <- sl.s.Close()
+		}(sl)
+	}
+	for range tps.children {
+		err := <-errChan
+		if errors.Is(err, http.ErrServerClosed) {
+			err = nil
+		}
+		if firstErr == nil {
+			firstErr = err
+		}
+		if err != nil {
+			log.Printf("[ERROR] %s", err)
+		}
+	}
+	return firstErr
+
+}
+
+// Serve - implements server.  The listener is ignored, but it
+// calls serve on the children
+func (tps *InetAfTCPProxyServer) Serve(_ net.Listener) error {
+	if len(tps.children) == 0 {
+		return fmt.Errorf("no children defined for listener")
+	}
+	errChan := make(chan error, len(tps.children))
```

Comment:
would be good to add a defer close here, too.

Reply:
see above